compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -147,93 +147,82 @@ LogicalResult ScatterOp::verify() {
   }
 
   auto indicesType = getIndicesType();
-  if (indicesType.getRank() < 2 ||
+  if (indicesType.getRank() < 1 ||
       !isa<IntegerType>(indicesType.getElementType())) {
-    return op->emitOpError("expected indices to be of rank 2 or greater and of "
+    return op->emitOpError("expected indices to be of rank 1 or greater and of "
                            "integer element type");
   }
-  auto indexDepth = getIndexDepth();
-  if (ShapedType::isDynamic(indexDepth)) {
-    return op->emitOpError("expected index depth is static");
-  }
 
   ArrayRef<int64_t> dimMap = getDimensionMap();
-  if (dimMap.size() != indexDepth) {
-    return op->emitOpError("invalid number of dimension map entries");
-  }
-
-  auto originalType = getOriginalType();
   if (failed(isPermSequence(
           [&]() { return this->emitOpError("dimension map is invalid."); },
           dimMap))) {
     return failure();
   }
 
-  if (indexDepth > originalType.getShape().size()) {
-    return op->emitOpError(
-        "index depth is greater than the rank of the original value");
+  if (dimMap.size() == 0) {
+    return op->emitOpError("dimension map must have at least one element");
   }
 
+  const size_t indexDepth = getIndexDepth();
+  auto originalType = getOriginalType();
   auto updateType = getUpdateType();
-  auto batchRank = indicesType.getRank() - 1;
+  const auto originalSliceRank = originalType.getRank() - indexDepth;
+  if (originalSliceRank < 0) {
+    return op->emitOpError(
+        "expected original rank to be greater or equal to index depth");
+  }
+  if (updateType.getRank() < originalSliceRank) {
+    return op->emitOpError(
+        "expected update to be at least the rank of non indexed original dims");
+  }
+  const size_t batchRank = updateType.getRank() - originalSliceRank;
+
+  if (updateType.getRank() - batchRank != originalType.getRank() - indexDepth) {
+    return op->emitOpError("expected rank of update value - batch rank to be "
+                           "equal to rank of original value - index depth");
+  }
   if (updateType.getRank() < batchRank) {
     return op->emitOpError("expected update value to be of rank greater than "
                            "or equal to rank(indices) - 1")
            << batchRank;
   }
 
-  // Validate the shape of indices and update value match for the first
-  // `batchRank` dims.
-  auto [indicesIt, updateIt] =
-      llvm::mismatch(indicesType.getShape().take_front(batchRank),
-                     updateType.getShape().take_front(batchRank));
-  if (indicesIt != indicesType.getShape().take_front(batchRank).end()) {
-    return op->emitOpError(
-               "mismatch in shape of indices and update value at dim#")
-           << (indicesIt - indicesType.getShape().begin());
+  if (indicesType.getRank() != batchRank &&
```

Comment:
We can be a bit stricter here by only allowing equality if `indexDepth == 0`
```suggestion
  if ((indicesType.getRank() != batchRank && indexDepth == 0) &&
```

Reply:
Good catch, this found some tests that were using invalid ops